compiler/rustc_typeck/src/coherence/inherent_impls_overlap.rs

```diff
@@ -113,9 +122,21 @@ impl ItemLikeVisitor<'v> for InherentOverlapChecker<'tcx> {
                 let ty_def_id = self.tcx.hir().local_def_id(item.hir_id);
                 let impls = self.tcx.inherent_impls(ty_def_id);
 
-                for (i, &impl1_def_id) in impls.iter().enumerate() {
-                    for &impl2_def_id in &impls[(i + 1)..] {
-                        if self.impls_have_common_items(impl1_def_id, impl2_def_id) {
+                // If there is only one inherent impl block,
+                // there is nothing to overlap check it with
+                if impls.len() <= 1 {
+                    return;
+                }
+
+                let impls_items = self.tcx.arena.alloc_from_iter(
+                    impls
+                        .iter()
+                        .map(|impl_def_id| (impl_def_id, self.tcx.associated_items(*impl_def_id))),
```

Comment:
Do we really need to allocate this on an arena? I guess it depends on whether the items are drop or not, but this may allocate a vector anyway and this way we don't really free up the used memory. I wonder if others have an opinion on this.

Reply:
Good point that we are allocating a vector anyways... and doing some cloning to get it into the arena. The arena uses smallvec internally as we both know so maybe it's best to use that one.